spec/order_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'order'
+
+describe Order do
+  let(:menu) { double :menu }
+  subject { described_class.new(menu: menu) }
+
+  before {
+    allow(menu).to receive_message_chain(:dishes, :key?).and_return true
+  }
+  
+  describe "#add" do
+    before { subject.add("Half Chicken", 2) }
+
+    it "should add several dishes to the current order" do
+      expect(subject.current).to eq "Your current order:\n2 x Half Chicken"
+    end
+
+    it "should raise error if item is not found on the menu" do
+      allow(menu).to receive_message_chain(:dishes, :key?).and_return false
+      expect { subject.add("Noodles", 1) }.to raise_error("Item not found on menu")
+    end
+  end
+
+  describe "#total" do
+    before { subject.add("Half Chicken", 2) }
+
+    it "should call dishes on Menu obejct" do
+      expect(menu).to receive_message_chain(:dishes, :[], :*)
+      subject.total
+    end
+
+    it "should return the total cost of the order" do
+      allow(menu).to receive_message_chain(:dishes, :[], :*).and_return(16)
```

Comment:
This stubbing is a bit too clever — you're not just mocking `menu`, but also starting to interfere with the way `Order` does its work. I.e., you could change the way `Order` calculates the product and this test would fail despite the fact that `Order` still works fine.

You probably ought to say `allow(menu).to receive(:dishes).and_return([{"Chicken Wing" => 16}])` or something similar, and let `total` do the work.